docs/content/integrations/airlift/operator-migration/python-operator.mdx

```diff
@@ -0,0 +1,75 @@
+# Migrating the PythonOperator to Dagster
+
+In this page, we'll explain migrating an Airflow `PythonOperator` to Dagster.
+
+### Background
+
+In Airflow, the `PythonOperator` runs arbitrary python functions. For example, you might have a task that runs function `write_to_db`, which combs a directory for files, and writes each one to a db table.
+
+```python file=/integrations/airlift/operator_migration/python_operator.py startafter=start_op endbefore=end_op
+from airflow.operators.python import PythonOperator
+
+
+def write_to_db() -> None:
+    for raw_file in RAW_DATA_DIR.iterdir():
+        df = contents_as_df(raw_file)
+        upload_to_db(df)
+
+
+PythonOperator(python_callable=write_to_db, task_id="db_upload", dag=...)
+```
+
+### Dagster equivalent
+
+The Dagster equivalent is instead to construct a <PyObject object="asset" module="dagster"/> or <PyObject object="multi_asset" module="dagster"/>-decorated function, which materializes assets corresponding to what your python function is doing.
+
+```python file=/integrations/airlift/operator_migration/pyop_multi_asset_complete.py startafter=start_asset endbefore=end_asset
+from dagster import asset
+
+
+@asset(key=TABLE_URI)
+def write_to_db() -> None:
+    for raw_file in RAW_DATA_DIR.iterdir():
+        df = contents_as_df(raw_file)
+        upload_to_db(df)
+```
+
+### Migrating the operator
+
+Migrating the operator breaks down into a few steps:
+
+1. Make a shared library available to both Airflow and Dagster with your python function.
+2. Writing an `@asset`-decorated function which runs the python function shared between both modules.
+3. Using `dagster-airlift` to proxy execution of the original task to Dagster.
+
+### Step 1: Building a shared library
+
+We recommend a monorepo setup for migration; this allows you to keep all your code in one place and easily share code between Airflow and Dagster, without complex CI/CD coordination.
+
+First, we recommend factoring out a shared package to be available to both the Dagster runtime and the Airflow runtime which contains your python function. The process is as follows:
+
+1. Scaffold out a new python project which will contain your shared infrastructure.
+2. Ensure that the shared library is available to both your Airflow and Dagster deployments. This can be done by adding an editable requirement to your `setup.py` or `pyproject.toml` file in your Airflow/Dagster package.
+3. Include the python dependencies relevant to your particular function in your new package. Write your python function in the shared package, and change your Airflow code to import the function from the shared library.
```

Comment:
Could we have a before and after example? Maybe by comparing the structure of the repo?

I think the idea is good, but these steps may be a bit vague for some users.

Reply:
good idea. Added a before and after example.